### PR TITLE
p2p: define p2p proto msgs

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

define p2p messages in proto

### CODE REVIEW

File: proto/p2p/types.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  string               network          = 4;
+  string               version          = 5;
+  bytes                channels         = 6;
+  string               moniker          = 7;
+  DefaultNodeInfoOther other            = 8 [(gogoproto.nullable) = false];
+}
+
+message DefaultNodeInfoOther {
+  string tx_index    = 1;
+  string rpc_address = 2 [(gogoproto.customname) = "RPC_ADDRESS"];
```
Comment (melekes) [inline proto/p2p/types.proto:34]: why all caps?

File: proto/p2p/conn_msgs.proto [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import "third_party/proto/gogoproto/gogo.proto";
+
+message PacketPing {}
+
+message PacketPong {}
+
+message PacketMsg {
+  int32 channel_id = 1 [(gogoproto.customname) = "ChannelID"];
+  int32 eof        = 2 [(gogoproto.customname) = "EOF"];
+  bytes byte       = 3;
```
Comment (melekes) [inline proto/p2p/conn_msgs.proto:15]:
```suggestion
  bytes data       = 3;
```

?

Comment (marbar3778) [reply proto/p2p/conn_msgs.proto]: ah was trying to keep parity with the go type, but data is more explanatory than` bytes` will rename the go type as well when the reactor is transitioned

File: proto/p2p/types.proto [hunk 3]
```diff
@@ -31,5 +31,5 @@ message DefaultNodeInfo {
 
 message DefaultNodeInfoOther {
   string tx_index    = 1;
-  string rpc_address = 2 [(gogoproto.customname) = "RPC_ADDRESS"];
+  string rpc_address = 2 [(gogoproto.customname) = "RPC_Adddress"];
```
Comment (melekes) [inline proto/p2p/types.proto:34]:
```suggestion
  string rpc_address = 2 [(gogoproto.customname) = "RPCAddress"];
```

**Pull request merged**